### DISCUSSION TITLE
Pydantic BaseSettings hot-reload

### DISCUSSION THREAD

**Discussion by elpapi42** [pydantic/pydantic#3048] [Question] [answered]
Hello!

Is there a way to setup BaseSetting to reload the env vars automatically, this allow for a more streamlined way of change env vars as required, and the service having fresh values everytime

Thanks!

**Comment by lesnik512** [comment, marked as answer]
I believe that's not feasible, because AFAIK you can't change envs of running process from outside.

You can change values of your settings global object on the fly based on some event. For example you can periodically read envs from some files. But still it seems like a bad pattern, because it's creates redundant complications and limitations.
